packages/e2e-tests/src/GPG.ts

```diff
@@ -0,0 +1,220 @@
+import { InputFactory } from './InputFactory';
+import child_process, { type SpawnSyncOptionsWithStringEncoding } from 'node:child_process';
+import fs from 'node:fs';
+import zxcvbn from 'zxcvbn';
+import * as core from '@actions/core';
+import { absPath, command, logSpawnSyncError } from './utilities';
+
+class GPG {
+	private readonly groupName: string = 'GPG';
+
+	constructor(private readonly inputs: InputFactory) {}
```

Comment:
not doing anything with `inputs`?
doesn't this need to be something like:

```js
	constructor(private readonly inputs: InputFactory) {
		this.inputs = inputs;
	}
```

or is this some fancy new JS voodoo you youngin's are using to cut corners?
back in MY day...

Reply:
It's a constructor promotion [similar to PHP](https://www.php.net/manual/en/migration80.new-features.php). This is a TypeScript specific feature called [parameter properties](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties):

> TypeScript offers special syntax for turning a constructor parameter into a class property with the same name and value

In terms of usage, this property is used here:

https://github.com/eventespresso/actions/blob/751d4a85eac61054f446442926dc0354c55bef14/packages/e2e-tests/src/GPG.ts#L145

https://github.com/eventespresso/actions/blob/751d4a85eac61054f446442926dc0354c55bef14/packages/e2e-tests/src/GPG.ts#L168

Reply:
oh cool, did not know that (i don't normally use classes in TS)